internal/course/database/course_queries.go

```diff
@@ -1,27 +1,39 @@
 package database
 
+import (
+	"fmt"
+)
+
 const (
-	createCourse       = "create course"
-	deleteCourse       = "delete course by uuid"
-	getCourse          = "get course by uuid"
-	listCourse         = "list course"
-	updateCourseByID   = "update course by uuid"
-	updateCourseByCode = "update course by code"
+	returningColumns = "uuid, code, name, underline, image, image_cover, excerpt, description, created_at, updated_at"
+	// CREATE.
+	createCourse = "create course"
+	// READ.
+	listCourse = "list course"
+	getCourse  = "get course by uuid"
+	// UPDATE.
+	updateCourse = "update course by uuid"
+	// DELETE.
+	deleteCourse = "delete course by uuid"
 )
 
 func queriesCourse() map[string]string {
 	return map[string]string{
-		createCourse: `INSERT INTO 
-    		courses (code, name, underline, image, image_cover, excerpt, description)
-			VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *`,
+		// CREATE.
+		createCourse: fmt.Sprintf(`INSERT INTO courses (
+				code, name, underline, image, image_cover, excerpt, description
+			) VALUES ($1, $2, $3, $4, $5, $6, $7)
+			RETURNING %s`, returningColumns),
+		// READ.
+		listCourse: fmt.Sprintf("SELECT %s FROM courses WHERE deleted_at IS NULL", returningColumns),
+		getCourse:  fmt.Sprintf("SELECT %s FROM courses WHERE uuid = $1 AND deleted_at IS NULL", returningColumns),
```

Comment:
If we plan on limiting this by active/archived, I suggest creating a `get_active` and `get_archived`, or you can remove the delete_at null and let this be populated via parameter

Reply:
https://github.com/sumelms/microservice-course/pull/19#discussion_r1520508615

https://github.com/sumelms/microservice-course/pull/19#discussion_r1520517473

Reply:
@rluders @arianerocha  additional comments?